File: Data Structures & Algorithms/linked-list-cycle-detection/submission-2.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {boolean}
     */
    hasCycle(head) {
        //1. using set
        if(!head) return false;
        let set=new Set(), curr=head;
        while(curr && curr.next){
            if(set.has(curr)) return true;
            set.add(curr);
            curr=curr.next;
        }
        return false;
    }
}

File: Data Structures & Algorithms/remove-linked-list-elements/submission-2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if(head == null) return head;

        //removing target from start of the list.
        while(head!=null && head.val == val){
            head = head.next;
        }
        ListNode curr = head, prev=null;
        while(curr!=null){
            if(curr.val == val){
                if(curr.next!=null){
                    prev.next = curr.next;
                }
                else prev.next = null;
            }
            else prev = curr;
            curr = curr.next;
        }       
        return head;
    }
}